src/code/z_actor.c

```diff
@@ -1524,22 +1524,33 @@ s32 func_8002F0C8(Actor* actor, Player* player, s32 flag) {
     return false;
 }
 
-u32 Actor_ProcessTalkRequest(Actor* actor, PlayState* play) {
-    if (actor->flags & ACTOR_FLAG_8) {
-        actor->flags &= ~ACTOR_FLAG_8;
+u32 Actor_AcceptTalkRequest(Actor* actor, PlayState* play) {
+    if (actor->flags & ACTOR_FLAG_TALK_REQUESTED) {
+        actor->flags &= ~ACTOR_FLAG_TALK_REQUESTED;
         return true;
     }
 
     return false;
 }
 
-s32 func_8002F1C4(Actor* actor, PlayState* play, f32 arg2, f32 arg3, u32 exchangeItemId) {
+/**
+ * This function covers offering the ability to `Talk` with the player.
+ * Passing an exchangeItemId (see `ExchangeItemID`) allows the player to also use the item to initiate the
+ * conversation.
+ *
+ * This function carries a talk exchange offer to the player actor if context allows it (e.g. the player is in range
+ * and not busy with certain things). The player actor performs the offered action itself.
+ *
```

Comment:
What does "perform the offered action" mean here? I dont really see any of this as offering an action, so maybe we need to clean that part up a bit

Reply:
Maybe its meant to convey that player handles the rest of the interaction from here. Can change the wording to be like that, or simply remove the sentence possibly

Reply:
Yeah, the idea is that player will then accept the offer and continues the processing, and that it's not done here. But then once the player returns the request, the actor then needs to do it's own work with progressing the talk, so it's a bit of back and forth. Because of that, I just ended up removing this line